iceoryx_examples/icehello/README.md

```diff
@@ -0,0 +1,173 @@
+# icehello
+
+## Introduction
+
+This example demonstrates a basic data transmission with zero-copy inter-process communication (IPC).
+It provides a publisher and a subscriber application.
+
+## Expected output
+
+<!-- Add asciinema link here -->
+
+## Code walkthrough
+
+### Publisher
+
+At first, we need to define what kind of topic the publisher and subscriber application will be exchanging:
+
+```cpp
+struct RadarObject
+{
+    RadarObject() noexcept
+    {
+    }
+    RadarObject(double x, double y, double z) noexcept
+        : x(x)
+        , y(y)
+        , z(z)
+    {
+    }
+    double x = 0.0;
+    double y = 0.0;
+    double z = 0.0;
+};
+```
+
+It is included by:
+
+```cpp
+#include "topic_data.hpp"
+```
+
+Next up, we include the publisher and the runtime:
+
+```cpp
+#include "iceoryx_posh/popo/publisher.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+```
+
+For the communication with the daemon RouDi, a runtime object is created. We use a unique string for identifying our application:
+
+```cpp
+constexpr char APP_NAME[] = "iox-cpp-publisher-helloworld";
+iox::runtime::PoshRuntime::initRuntime(APP_NAME);
+```
+
+Up next, we create a publisher instance for our charming struct, notice the topic type is passed as a template parameter:
+
+```cpp
+iox::popo::Publisher<RadarObject> publisher({"Radar", "FrontLeft", "Object"});
+```
+
+The three strings passed as parameter to the constructor of `iox::popo::Publisher` are of the type
```

Comment:
The three strings that are passed as parameters to the constructor of `iox::popo::Publisher` form our `capro::ServiceDescription`.